crates/apollo-encoder/src/value.rs

```diff
@@ -0,0 +1,114 @@
+use std::fmt;
+
+/// The __value type represents available values you could give as an input.
+///
+/// *Value*:
+///     Variable | IntValue | FloatValue | StringValue | BooleanValue | NullValue | EnumValue | ListValue | ObjectValue
+///
+/// Detailed documentation can be found in [GraphQL spec](https://spec.graphql.org/October2021/#Value).
+#[derive(Debug, PartialEq, Clone)]
+pub enum Value {
+    /// Name of a variable example: `varName`
+    Variable(String),
+    /// Int value example: `7`
+    Int(i64),
+    /// Float value example: `25.4`
+    Float(f64),
+    /// String value example: `"My string"`
+    String(String),
+    /// Boolean value example: `false`
+    Boolean(bool),
+    /// Null value example: `null`
+    Null,
+    /// Enum value example: `"VARIANT_EXAMPLE"`
+    Enum(String),
+    /// List value example: `[1, 2, 3]`
+    List(Vec<Value>),
+    /// Object value example: `{ first: 1, second: 2 }`
+    Object(Vec<(String, Value)>),
+}
+
+impl fmt::Display for Value {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Variable(v) => write!(f, "${v}"),
+            Self::Int(i) => write!(f, "{i}"),
+            Self::Float(fl) => write!(f, "{fl}"),
+            Self::String(s) => {
+                if s.contains('"') {
```

Comment:
it's also a block string character if it contains `\n` or `\r`
```suggestion
                if s.contains('"') | s.contains('\n') | s.contains('\r') {
```